core/src/main/scala/org/apache/predictionio/controller/PAlgorithm.scala

```diff
@@ -115,15 +115,12 @@ abstract class PAlgorithm[PD, M, Q, P]
     algoParams: Params,
     bm: Any): Any = {
     val m = bm.asInstanceOf[M]
-    if (m.isInstanceOf[PersistentModel[_]]) {
-      if (m.asInstanceOf[PersistentModel[Params]].save(
-        modelId, algoParams, sc)) {
-        PersistentModelManifest(className = m.getClass.getName)
-      } else {
-        ()
-      }
-    } else {
-      ()
+    m match {
+      case m: PersistentModel[Params] @unchecked =>
+        if(m.save(modelId, algoParams, sc)){
+          PersistentModelManifest(className = m.getClass.getName)
+        } else ()
+      case _ => ()
```

Comment:
> This code will be simpler to read by joining the if guard with the case statement.

Yes, but I want to keep the form of code as same as other algorithms.

> Which is correct?

I don't know. In any case, I think that I shouldn't modify current behavior in this pull request because this pull request is for refactoring.